python/voxel_mapper.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+import sys
+import os
+import nibabel as nib
+import  numpy as np
+import numbers
+import argparse
+
+
+        
+def mapper(bin_size, data_max, data_min):
+    for line in sys.stdin:
+            line = line.strip()
+            line = line.split()
+            slice_file = os.path.join(line[0])
+            
+
+            #open nifti image
+            slice = nib.load(slice_file)
+            data = slice.get_data().flat
+     
+            histogram = {}
+            
+
+            #iterate throught the voxels and add them to dictionary with an initial count of 1
+            #if same voxel value is encountered again, increment the count of the voxel value by one
+            for value in data:
+                try:
+                    voxel_value = float(value.item())
+                except:
+                    print('Error: Voxel value could not be cast to a float')
+                    sys.exit(1)
+                #dictionary key will be in the form of (min, max) tuple. All values will be in min/max range, excluding the max, which will be the min value of the proceeding bin
+                try:
+                    
+                    if (voxel_value < data_min or voxel_value > data_max):
+                        continue
+
+                    remainder = (voxel_value - data_min) % bin_size if voxel_value >= 0 else -(voxel_value - data_min) % bin_size
```

Comment:
So the remainder will be negative in case voxel_value is negative. I think it should always be positive, i.e., you should remove the if-else condition.